src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -228,10 +228,11 @@ static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerabl
                 operationKey = OperationEqualsInstance;
             }
             // Analyze argument operation, potentially 0.Equals(obj.Count()).
-            else if (parentOperation is IArgumentOperation argumentOperation)
+            else if (parentOperation is IArgumentOperation argumentOperation &&
+                argumentOperation.Parent is IInvocationOperation invocation)
```

Comment:
What happens if there is a conversion or parenthesis in the argument? e.g: `0.Equals((uint)obj.Count())`.

Reply:
nit: 
```suggestion
                argumentOperation.Parent is IInvocationOperation invocationOperation)
```

Reply:
I will make some test and update the code accordingly. Thanks!

Reply:
> What happens if there is a conversion or parenthesis in the argument?

IArgumentOperation is still the highest node: https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Generated/Operations.Generated.cs#L646

Reply:
That's interesting, we have a diagnostic raised suggesting to use `.Any()` instead of `Count()` while we should not in this case. `0.Equals((uint)obj.Count())`

Reply:
Can you share your scenario? I can't repro it.
```cs
[Fact]
public Task CSharpTest()
    => VerifyCS.VerifyAnalyzerAsync(
    @"using System.Linq;

    class C
    {
        private string _str;
        public int GetLength() => 0.Equals((uint)_str.Count()) ? 
            0 : _str.Length;
    }", VerifyCS.Diagnostic(UseCountProperlyAnalyzer.CA1829).WithLocation(6, 46).WithArguments("Length"));
}
```

Reply:
@Evangelink Are you referring to test `PropertyAccessParentIsNotAlwaysDirectlyTheInvocation`?